drivers/flash/flash_stm32.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2017 BayLibre, SAS.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <flash.h>
+#include <init.h>
+#include <soc.h>
+
+#include <flash_stm32.h>
+
+#define STM32_FLASH_TIMEOUT	((u32_t) 0x000B0000)
+
+int flash_stm32_wait_flash_idle(struct flash_stm32_priv *p)
+{
+#if defined(CONFIG_SOC_SERIES_STM32F4X)
+	volatile struct stm32f4x_flash *regs = p->regs;
```

Comment:
+1 (removing my redundant comment).
And if this is the right explanation, it would be worse aligning the structures one way or the other.